test.py

```diff
@@ -0,0 +1,191 @@
+    => $F in {A, Q, g, w} (0bxx0000)"""
+
+    charsAll = ''.join(sorted('ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_'))
+    # 0 - 31: 0b0xxxxx
+    charsC = ''.join(sorted('ABCDEFGHIJKLMNOPQRSTUVWXYZabcdef'))
+    # 16, 17, 18, 19: 0b0100xx
+    charsD = ''.join(sorted('QRST'))
+    # 2, 6, 10, 14, 18, 22, 26, 30, 34, 38, 42, 46, 50, 54, 58, 62: 0bxxxx10
+    charsE = ''.join(sorted('CGKOSWaeimquy26-'))
+    # 0, 16, 32, 48: 0bxx0000
+    charsF = ''.join(sorted('AQgw'))
+    expected = [charsC, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsD, charsAll, charsE, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsF]
+    spreadTest(slugid.nice, expected)
+
+
+def testSpreadV4():
+    """ This test is the same as niceSpreadTest but for slugid.v4() rather than
+    slugid.nice(). The only difference is that a v4() slug can start with any of
+    the base64 characters since the first six bits of the uuid are random."""
+
+    charsAll = ''.join(sorted('ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_'))
+    # 16, 17, 18, 19: 0b0100xx
+    charsD = ''.join(sorted('QRST'))
+    # 2, 6, 10, 14, 18, 22, 26, 30, 34, 38, 42, 46, 50, 54, 58, 62: 0bxxxx10
+    charsE = ''.join(sorted('CGKOSWaeimquy26-'))
+    # 0, 16, 32, 48: 0bxx0000
+    charsF = ''.join(sorted('AQgw'))
+    expected = [charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsD, charsAll, charsE, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsAll, charsF]
+    spreadTest(slugid.v4, expected)
+
+
+def spreadTest(generator, expected):
+    """ `spreadTest` runs a test against the `generator` function, to check that
+    when calling it 64*40 times, the range of characters per string position it
+    returns matches the array `expected`, where each entry in `expected` is a
+    string of all possible characters that should appear in that position in the
+    string, at least once in the sample of 64*40 responses from the `generator`
+    function """
+    # k is an array which stores which characters were found at which
+    # positions. It has one entry per slugid character, therefore 22 entries.
+    # Each entry is a dict with a key for each character found, and its value
+    # as the number of times that character appeared at that position in the
+    # slugid in the large sample of slugids generated in this test.
+    k = [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}]
+
+    # Generate a large sample of slugids, and record what characters appeared
+    # where...  A monte-carlo test has demonstrated that with 64 * 20
+    # iterations, no failure occurred in 1000 simulations, so 64 * 40 should be
+    # suitably large to rule out false positives.
+    for i in range(0, 64 * 40):
```

Comment:
If we are sticking with Python 2, might be better use [xrange](https://docs.python.org/2/library/functions.html#xrange).